### PR TITLE
Instruction for debugging analyzer with runtime repo projects

### PR DESCRIPTION
**Pull request by buyaa-n**

### CODE REVIEW

File: docs/NetCore_GettingStarted.md [hunk 1]
```diff
@@ -70,3 +71,12 @@
     - `.\artifacts\bin\AnalyzerRunner\Release\netcoreapp3.1\AnalyzerRunner.exe ..\roslyn-analyzers\artifacts\bin\Microsoft.NetCore.Analyzers.Package\Debug\netstandard2.0 .\Roslyn.sln /stats /concurrent /a AnalyzerNameToTest /log Output.txt`
     - Do not forget change value after `/a` option with your testing analyzer name.
 The diagnostics reported by the analyzer will be listed in Output.txt.
+
+## Debugging analyzer with runtime repo projects
+1. Copy over debug build of analyzer assemblies on top of NetAnalyzers nuget package in your packages folder. (Instructions are same as the step 1 and 2 of [Testing against the Runtime repo step](#Testing-against-the-Runtime-and-Roslyn-analyzers-repo))
```
Comment (Evangelink) [inline docs/NetCore_GettingStarted.md:76]:
```suggestion

1. Copy over debug build of analyzer assemblies on top of NetAnalyzers nuget package in your packages folder. (Instructions are same as the step 1 and 2 of [Testing against the Runtime repo step](#Testing-against-the-Runtime-and-Roslyn-analyzers-repo))
```

File: docs/NetCore_GettingStarted.md [hunk 2]
```diff
@@ -70,3 +71,12 @@
...
+2. Start VS and open a project you want to debug
+3. Note the process ID for `ServiceHub.RoslynCodeAnalysisService.exe` corresponding to that VS instance
+4. Open another VS instance for `RoslynAnalyzers.sln` and set breakpoints in the analyzer solution where you want to debug
+5. Attach to above process ID with the RoslynAnalyzers debugger: `Debug -> Attach to Process...`
+6. Start typing in the other project, the breakpoints should hit
+    - If breakpoints are not hitting then the RoslynAnalyzers.sln build might not the same as the build you copied to the step 1 repeat the step again or check if you copied into the correct path
```
Comment (Evangelink) [inline docs/NetCore_GettingStarted.md:82]:
```suggestion
    - If breakpoints are not hitting then the RoslynAnalyzers.sln build might not the same as the build you copied to the step 1 repeat the step again or check if you copied into the correct path
    
```

File: docs/NetCore_GettingStarted.md [hunk 3]
```diff
@@ -39,7 +39,7 @@
 
 1. Navigate to the root of the Roslyn-analyzers repo and run these commands:
     - `cd roslyn-analyzers`
-    - `set RUNTIMEPACKAGEVERSION=3.0.0`
+    - Set RUNTIMEPACKAGEVERSION variable with the version which ever the runtime/roslyn-analyzers repo is using: `set RUNTIMEPACKAGEVERSION=5.0.0`
```
Comment (Evangelink) [inline docs/NetCore_GettingStarted.md:42]: Could you point to a file or something where we could get the info?

Comment (buyaa-n) [reply docs/NetCore_GettingStarted.md]: Sure, added a link

**Review by Evangelink — APPROVED** [review]

File: docs/NetCore_GettingStarted.md [hunk 4]
```diff
@@ -70,3 +71,14 @@
     - `.\artifacts\bin\AnalyzerRunner\Release\netcoreapp3.1\AnalyzerRunner.exe ..\roslyn-analyzers\artifacts\bin\Microsoft.NetCore.Analyzers.Package\Debug\netstandard2.0 .\Roslyn.sln /stats /concurrent /a AnalyzerNameToTest /log Output.txt`
     - Do not forget change value after `/a` option with your testing analyzer name.
 The diagnostics reported by the analyzer will be listed in Output.txt.
+
+## Debugging analyzer with runtime repo projects
+
+1. Copy over debug build of analyzer assemblies on top of NetAnalyzers nuget package in your packages folder. (Instructions are same as the step 1 and 2 of [Testing against the Runtime repo step](#Testing-against-the-Runtime-and-Roslyn-analyzers-repo))
+2. Start VS and open a project you want to debug
+3. Note the process ID for `ServiceHub.RoslynCodeAnalysisService.exe` corresponding to that VS instance
```
Comment (mavasani) [inline docs/NetCore_GettingStarted.md:79]: Do we want to add a section for debugging code fix provider as well? The code fix providers will execute in `devenv.exe`, so the step 3. here would be slightly different. If you do not want to duplicate the whole section, you can just add another line to this third step about code fixers.

**Review by mavasani — APPROVED** [review]

File: docs/NetCore_GettingStarted.md (hunk 4 above)
Comment (buyaa-n) [reply docs/NetCore_GettingStarted.md:79]: > Do we want to add a section for debugging code fix provider as well? The code fix providers will execute in devenv.exe, so the step 3. here would be slightly different. If you do not want to duplicate the whole section, you can just add another line to this third step about code fixers.

I would love to add that, but i don't know how it works, never tried

Comment (mavasani) [reply docs/NetCore_GettingStarted.md:79]: Ah, you can just add a line to step 3 that if you want to debug the CodeFixProvider corresponding to the analyzer, then you will need to attach a different process, i.e. devenv.exe. Code fixes and analyzers run in different processes.

File: docs/NetCore_GettingStarted.md [hunk 5]
```diff
@@ -70,3 +71,16 @@
...
+  - If you are using `Visual Studio` older than version `16.8 Preview2` then analyzers run in `devenv.exe`, you will need to attach that process instead
+  - Code fixes and analyzers run in different processes. If you want to debug the CodeFixProvider corresponding to the analyzer, attach `devenv.exe` instead.
```
Comment (buyaa-n) [inline docs/NetCore_GettingStarted.md:81]:
```suggestion
    - If you are using `Visual Studio` older than version `16.8 Preview2` then analyzers run in `devenv.exe`, you will need to attach that process instead
    - Code fixes and analyzers run in different processes. If you want to debug the CodeFixProvider corresponding to the analyzer, attach `devenv.exe` instead. 
```

**Review by mavasani — APPROVED** [review]

**Pull request merged**